### ISSUE TITLE
iOS Open Database from AppGroup container

### ISSUE THREAD

**Issue by accidbright** [tekartik/sqflite#517]
I'm not able to find anywhere information about how to open database that is placed not in Documents directory of app sandbox in iOS (that is default location). I need to open database from AppGroup container. Is there a possibility to customize database path?

**Comment by alextekartik** [comment]
I'm not familiar with the AppGroup container but if I understand correctly, you can specify an absolute path when opening a database. The databases path (which is the documents directory on iOS) is only used when you specify a relative database path.

Make sure to create the parent directory though.

**Comment by accidbright** [comment]
@alextekartik thanks for the reply.
As far as I can understand, absolute path can be specified in Dart code, right? But AppGroup container path is not static path, it can be different on different devices (same as path to Documents folder). It can be gotten the same way as path to Documents directory (but calling other functions in native code). 

So it looks to me like the workaround is like this:
1. Get path to container in native code
2. Implement custom plugin with method like `getAppGroupDatabasePath`
3. Set this received path as absolute path to database

Overall looks like workaround, but I don't like the step 2, because it required some additional work and looks like the plugin does not support this feature out-of-the-box. Or do I miss something else?

**Comment by alextekartik** [comment]
Thanks for the explanation. Maybe getting such path should be part of `path_provider` as I don't think that this is database specific.

**Comment by accidbright** [comment]
No, `path_provider` doesn't contain this path (just checked). To have this path you should have special capability in your app. The path can be gotten by unique AppGroup name registered at the developer portal. AppGroups are usually used for sharing data between apps, app and widget or app and watch app.
